### ISSUE TITLE
Loading PHP mailer globaly on one page

### ISSUE THREAD

**Issue by Oalddesign** [PHPMailer/PHPMailer#1944]
Hey, first of all I have to see i have been using this code for sometime now and it's really great.

I am trying to load PHPmailer into a system and load it from the main system file which will load globally on all the files so i can use and send mails on all files.

So let say the main page loading everything is index.php and i load at the top of the document this code: 

    use PHPMailer\PHPMailer\PHPMailer;
    use PHPMailer\PHPMailer\SMTP;
    use PHPMailer\PHPMailer\Exception;

    require_once("include/phpmailer/autoload.php");

Now inside index.php i include other files that send mails. Let say one of the files is second.php and from there i send the mail.

I found that if i do not reuse this code at the top of second.php file I am keep getting an error:
    use PHPMailer\PHPMailer\PHPMailer;
    use PHPMailer\PHPMailer\SMTP;
    use PHPMailer\PHPMailer\Exception;

Is it possible that i have to keep using it at the top of each file i include even due i load them on the main file "index,php"? seems like it's very inefficient and messy as i have to load them at the top of the document and not with the send mail code. In the older versions of PHPMailer it wasn't required and it was easier to use.

Will appreciate if you can tell  me if there is a way to go around that instead of including the same code over and over again.
Thanks

**Comment by XL-2000** [comment]
Why not put:

`use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

require_once("include/phpmailer/autoload.php");`

into a separate container file "loadphpmailer.php"

and then 

`require_once("loadphpmailer.php");`

in index.php or where ever

**Comment by Oalddesign** [comment]
Hey thanks for your answer,
Loading it to a new different file doesn't fix my problem,

The ideal way of using it will be to load it once into index.php and have it work on all the other files.
By including it on different file i will still have load loadphpmailer.php on each file i use to send mails which make it a pain in the ass and it will probably cause problems in the future which changes.

The main idea is to load it only on the main index.php and have it work on all the included files on that page. Is it not something that possible with this one?

**Comment by Synchro** [comment]
You only need `use` statements in files that reference namespaced classes; if you’re only loading the class files, you don’t need them. Also note that use statements apply only to the file they appear in, so you can’t remove the repetition by putting them in an include file.

Overall it looks like you’re making a lot of extra work for yourself by not using composer.

**Comment by Oalddesign** [comment]
Hey thank you for your reply appreciate that,

I am what you will probably call a lightweight coder so to be honest i am not sure i was able to understand what you wrote above.

I did used composer to download the code into my computer and loaded the autoload.php like in your example. 

// Load Composer's autoloader
require 'vendor/autoload.php';

What do you mean by that? "if you’re only loading the class files" is there other file i can load and i won't use to declare "use" on the top of each page?

I have noticed that if i load "PHPMailer.php" directly i only need to declare "use PHPMailer\PHPMailer\PHPMailer;" and it doesn't have to be at the top of the page. Is that good practice or is it wrong.

Thanks

**Comment by Synchro** [comment]
`Use` statements are a way of importing or aliasing class names into your current namespace, which helps keep your code looking tidy. This:

    use PHPMailer\PHPMailer\PHPMailer;
    $mail = new PHPMailer(); 

Is exactly equivalent to:

    $mail = new PHPMailer\PHPMailer\PHPMailer();

Namespacing allows you to avoid class name clashes, so if you had some other library that contained a `PHPMailer` class, you can create aliases to remove ambiguity, like this:

    use PHPMailer\PHPMailer\PHPMailer;
    use some\other\PHPMailer as otherPHPMailer;
    $mail = new PHPMailer();
    $othermail = new otherPHPMailer();

The important thing to note here is that the `use` statements are completely separate from actually loading the class (with `require`/`include`), however, if you load the PHPMailer class without either aliasing it or using a fully-qualified class name, you would get an error if you tried to do `new PHPMailer()` because that class would not exist in your current namespace (which is probably the default "global" namespace).

My comment about only loading the classes is that if you're loading classes but not using them, you don't need `use` statements, that is, you can run `vendor/autoload.php`, and then you only need to have `use` statements for classes you're actually creating instances of in that file.

In short, `use` statements are entirely normal in modern PHP applications; it's convention to put them all at the top of your script (tools like phpcbf will move them there for you); so get used to them!

**Comment by Oalddesign** [comment]
Thanks a lot sir, appreciate the time you put into the detailed reply.
Hope you will have a great day
